core/common/src/MapLayerSettings.ts

```diff
@@ -349,24 +405,98 @@ export class MapLayerSettings {
+  }
 }
+/** Normalized representation of a [[ModelMapLayerProps]] for which values have been validated and default values have been applied where explicit values not defined.
+ * Model map layers are produced from models, typically from two dimensional geometry that may originate in a GIS system.
+ * One or more map layers may be included within [[MapImagerySettings]] object.
+ * @see [[MapImagerySettings]]
+ * @see [[ImageMapLayerSettings]] for image based map layer settings.
+ * @beta
+ */
+export class ModelMapLayerSettings extends MapLayerSettingsBase {
+  public readonly modelId: Id64String;
+  public override get source(): string { return this.modelId; }
+
+  /** @internal */
+  protected constructor(modelId: Id64String,  name: string, visible = true,
+    transparency: number = 0, transparentBackground = true, isBase = false) {
+    super(name, visible, transparency, transparentBackground, isBase);
+    this.modelId = modelId;
+  }
+  /** Construct from JSON, performing validation and applying default values for undefined fields. */
+  public static override fromJSON(json: ModelMapLayerProps): ModelMapLayerSettings {
+    const transparentBackground = (json.transparentBackground === undefined) ? true : json.transparentBackground;
+    return new this(json.modelId, json.name, json.visible, json.transparency, transparentBackground, json.isBase === true);
+  }
+  /** return JSON representation of this MapLayerSettings object */
+  public override toJSON(): ModelMapLayerProps {
+    const props = super.toJSON();
+    return { ... props, modelId: this.modelId };
+  }
+  /** Create a copy of this MapLayerSettings, optionally modifying some of its properties.
+   * @param changedProps JSON representation of the properties to change.
+   * @returns A MapLayerSettings with all of its properties set to match those of `this`, except those explicitly defined in `changedProps`.
+   */
+  public clone(changedProps: Partial<ModelMapLayerProps>): ModelMapLayerSettings {
+    return ModelMapLayerSettings.fromJSON(this.cloneProps(changedProps));
+  }
+  /** @internal */
+  protected override cloneProps(changedProps: Partial<ModelMapLayerProps>): ModelMapLayerProps {
+    const baseProps = super.cloneProps(changedProps);
+    return {... baseProps, modelId: undefined !== changedProps.modelId ? changedProps.modelId : this.modelId };
+  }
+  /** @internal */
+  public override displayMatches(other: MapLayerSettings): boolean {
+    if (other instanceof ImageMapLayerSettings || !super.displayMatches(other))
+      return false;
+
+    return this.modelId === other.modelId;
+  }
+  /** Return true if all sublayers are invisible (always false as model layers do not include sublayers). */
+  public get allSubLayersInvisible(): boolean {
+    return false;
+  }
+}
+
+/** Settings for a map layer, either [[ImageMapLayerSettings]]  or [[ModelMapLayerSettings]]
+ * @beta
+ */
+export type MapLayerSettings = ImageMapLayerSettings | ModelMapLayerSettings;
```

Comment:
Why isn't MapLayerSettings the base class for ImageMapLayerSettings and ModelMapLayerSettings - why do we need this extra union type plus MapLayerSettingsBase?

Reply:
Yes -- That is more straightforward and seems to work fine.   To get that to work I had to do same with MapLayersProps -- removing the union and having the MapImagerySettings just defined with the base class...   Is that right?  It makes it impossible to define MapImageryProps as follows as the url etc. are not in the base class.


    const mapImagery: MapImageryProps = {
      backgroundBase: ColorDef.red.tbgr,
      backgroundLayers: [{
        name: "x",
        url: "y",
        transparency: 0.5,
        formatId: "WMS",
        visible: true,
      }], 
    };

Reply:
I wasn't a fan of `MapLayerPropsBase` anyway. I can take a crack at cleaning this up if you like.

Reply:
I'll push my changes soon.

Reply:
Changes pushed... Please take a look at MapLayerProps/MapLayersPropsBase - Using base class to define MapImageryProps doesn't seem quite right (imodel.test.ts doesn't build).   DisplayStyleState.changeMapLayerProps still uses the union.